### DISCUSSION TITLE
Import config values from next.config.js

### DISCUSSION THREAD

**Discussion by nico-bachner** [vercel/next.js#23180] [Help]
Is there any way of importing configuration values from `next.config.js`?

For example, I'd like to access the i18n `defaultLocale` value: 
```cjs
// next.config.js
module.exports = {
    i18n: {
        locales: ["en", "da", "fr", "de", "lu"],
        defaultLocale: "en",
    },
}
```

I tried something along these lines, but I got a ton of errors.
```jsx
// lib/i18n.jsx
import nextConfig from "../next.config"

const i18nDefaultLocale = nextConfig.i18n.defaultLocale
```

Is there any way of accessing the `next.config.js` values, or is this something that is purposefully impossible?

**Comment by akellbl4** [comment]
I could suggest storing your `i18n` preferences in `json` file and load it in your code and config.
`require` works well for json, and next.js support import from `json` files
